gui/qt/history_list.py

```diff
@@ -184,11 +184,18 @@ def update_labels(self):
             return
         root = self.invisibleRootItem()
         child_count = root.childCount()
+        filtered_list = []
         for i in range(child_count):
             item = root.child(i)
             txid = item.data(0, Qt.UserRole)
             label = self.wallet.get_label(txid)
             item.setText(3, label)
+            filtered = run_hook("history_list_filter", self, item, label, multi=True) or []
```

Comment:
The "API" here is to send it an "h_item" as the 3rd argument to the hook, not a `QTreeListWidgetItem`.   An `h_item` is a 6-tuple of `tx_hash, height, conf, timestamp, value, balance` (see `history_list.py` class method `on_update`).

Right now nothing dereferences this argument but in the future that may not always be the case.

Reply:
I've done some changes to only run the hook on changed labels and also use h_item as parameter.

Reply:
Man this going to be slow too -- looping over all the history just to find 1 item?!

Reply:
You realize some wallets have like tens of thousands of items in their history right?  wallet.get_history() is not exactly a "cheap" call.

Reply:
I appreciate the effort to pass "good data" down but at this point I'd just take a `None` there and avoid this very expensive call...

Please -- just pass `None` there and avoid that call to `get_history` altogether. Maybe you can go to the 2 callsites in shuffle and fusion plugins and add a comment that h_item may be `None`... 

Otherwise I like the `item_changed` approach, with that 1 caveat that it need not be this expensive.